### PR TITLE
Refactoring to add HA integration code

### LINKED ISSUE #2: Feature: Home Assistant Integration
https://developers.home-assistant.io/docs/creating_component_index

### PR DESCRIPTION
**Pull request by titilambert**
This PR fixes #2 and #14 

Hello ! this PRs adds the HA integration !
So please feel free to test it ! and also don't forget to test also the `chihirosctl` CLI !
Thanks !

### CODE REVIEW

**Comment by titilambert** [comment]
CC @TheMicDiet @boominger @kingunlim

File: custom_components/chihiros/chihiros_led_control/chihirosctl.py [hunk 1]
```diff
@@ -41,25 +41,56 @@ async def _async_func() -> None:
...
-    table = Table("Name", "Address")
+    """List all bluetooth devices.
+
+    TODO: add an option to show only Chihiros devices
+    """
+    table = Table("Name", "Address", "Model")
     discovered_devices = asyncio.run(BleakScanner.discover(timeout=timeout))
     for device in discovered_devices:
-        table.add_row(device.name, device.address)
+        assert device.name is not None
```
Comment (TheMicDiet) [inline custom_components/chihiros/chihiros_led_control/chihirosctl.py:51]: if there are devices nearby without a name this command doesn't work anymore

Comment (TheMicDiet) [reply custom_components/chihiros/chihiros_led_control/chihirosctl.py:51]: maybe only print out devices with a name

**Review by TheMicDiet — CHANGES_REQUESTED** [review]
Hi, nice work! I couldn't try the HA part because I currently have no HA instance running. However, there are some issues with the CLI part.

File: custom_components/chihiros/chihiros_led_control/device/fallback.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+class Fallback(BaseDevice):
+    """Fallback device used when a device is not completely supported yet."""
+
+    _model_name = "fallback"
+    _model_code = ""
+    _colors: dict[str, int] = {
+        "red": 0,
+        "green": 1,
+        "blue": 2,
+    }
```
[the remark this answers is not in the document]
Comment (titilambert) [reply custom_components/chihiros/chihiros_led_control/device/fallback.py:15]: Hum ... white is also 0 right ?

File: custom_components/chihiros/chihiros_led_control/chihirosctl.py (hunk 1 above)
Comment (titilambert) [reply custom_components/chihiros/chihiros_led_control/chihirosctl.py:51]: I think the name is always set (and stay XXXXMACADDRESS) even when you change it in the app

File: custom_components/chihiros/chihiros_led_control/device/fallback.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (titilambert) [reply custom_components/chihiros/chihiros_led_control/device/fallback.py:16]: should be fixed

File: custom_components/chihiros/chihiros_led_control/chihirosctl.py (hunk 1 above)
Comment (TheMicDiet) [reply custom_components/chihiros/chihiros_led_control/chihirosctl.py:51]: This might be true for chihiros devices but not for every Bluetooth device

**Comment by titilambert** [comment]
@TheMicDiet it should be ready for review

**Review by TheMicDiet — APPROVED** [review]
LGTM

**Pull request merged**